clients/drcachesim/tracer/tracer.cpp

```diff
@@ -272,7 +354,9 @@ write_trace_data(void *drcontext, byte *towrite_start, byte *towrite_end)
     if (op_offline.get_value()) {
         per_thread_t *data = (per_thread_t *) drmgr_get_tls_field(drcontext, tls_idx);
         ssize_t size = towrite_end - towrite_start;
-        if (file_ops_func.handoff_buf != NULL) {
+        if (op_num_threads.get_value()) {
+            trace_buf_enqueue(data->file, towrite_start, size);
+        } else if (file_ops_func.handoff_buf != NULL) {
```

Comment:
So you've put a priority here: the handoff is going to be ignored if there's a thread pool.  That should be reflected in its docs.

Maybe we should just remove the handoff feature completely, since it's not needed anymore: it can be duplicated by using a thread pool and having file_ops_func.write do whatever you want.

I also think the thread pool should be the default.